pkg/socketenricher/tracer.go

```diff
@@ -66,6 +101,67 @@ func NewSocketEnricher() (*SocketEnricher, error) {
 	return se, nil
 }
 
+// Types returns the types and the BTF struct for the sockets_value map
+func (se *SocketEnricher) Types() ([]btf.Type, *btf.Struct, error) {
+	return se.types, se.valueBtfStruct, nil
+}
+
+func (se *SocketEnricher) generateTypes() error {
+	uint32T := btfhelpers.BtfInt(4, btf.Unsigned)
+	charT := btfhelpers.BtfInt(1, btf.Char)
+
+	types := []btf.Type{uint32T, charT}
+
+	// Look for the sockets_val btf structure to fill fixed members
+	srcBtfStruct := &btf.Struct{}
+	if err := se.spec.Types.TypeByName(SocketsValueName, &srcBtfStruct); err != nil {
+		return fmt.Errorf("getting BTF struct %q: %w", SocketsValueName, err)
+	}
+
+	members := []btf.Member{}
+	currentOffset := uint32(0)
+	for _, member := range srcBtfStruct.Members {
+		// stop when we find the first optional field, cwd.
+		if member.Name == "cwd" {
```

Comment:
Instead of hardcoding "cwd" as the first optional field, the struct could have a marker:

```c
struct sockets_value {
  //...mandatory fields...
  ...

  char optional_fields_start[0]; // zero-sized field

  //...optional fields...
  char cwd[123];
  ...
}
```

Reply:
Good idea, also worked for me!